src/VecSim/algorithms/brute_force/vector_block.h

```diff
@@ -16,7 +16,7 @@ struct VectorBlock;
 struct CompareByFirst {
     constexpr bool operator()(std::pair<float, labelType> const &a,
                               std::pair<float, labelType> const &b) const noexcept {
-        return a.first < b.first;
+        return (a.first != b.first) ? a.first < b.first : a.second < b.second;
```

Comment:
why don't we move this comparator to utils/common helper file?
this is the same implementation here and in hnswlib?

Reply:
This isn't exactly the same implementation, since in hnswlib this comparator is generic. If and when we refactor our hnswlib implementation to use `float` as data type instead of being generic, we should definitely move this to some utils file...